src/react-infinite.jsx

```diff
@@ -287,23 +287,27 @@ var Infinite = React.createClass({
           this.preservedScrollState);
       }
     }
-    if (React.Children.count(this.props.children) !== React.Children.count(prevProps.children)) {
+
+    const hasLoadedMoreChildren = React.Children.count(this.props.children) !== React.Children.count(prevProps.children);
+    if (hasLoadedMoreChildren) {
       var newApertureState = infiniteHelpers.recomputeApertureStateFromOptionsAndScrollTop(
         this.state,
         this.utils.getScrollTop()
       );
       this.setState(newApertureState);
     }
+
+    const isMissingVisibleRows = hasLoadedMoreChildren && !this.hasAllVisibleItems() && !this.state.isInfiniteLoading;
```

Comment:
@garetht I've looked at this further, and the predicate seems to be correct. I could move the logic into `componentWillReceiveProps` potentially if you would prefer?

Can you think of any edge-cases or such?